ledger/voters.go

```diff
@@ -249,3 +302,32 @@ func (vt *votersTracker) getVoters(r basics.Round) (*ledgercore.VotersForRound,
 
 	return tr, nil
 }
+
+func (vt *votersTracker) registerPrepareCommitListener(commitListener ledgercore.VotersCommitListener) {
+	vt.commitListenerMu.Lock()
+	defer vt.commitListenerMu.Unlock()
+
+	vt.commitListener = commitListener
```

Comment:
What would you recommend we do if it is already set?
return an error?

Reply:
What happens if it is being overwritten? Is it expected to be called multiple times? 
I think it will be a bug. In that case, an error is warranted.

Reply:
With the Start/Stop, my previous comment is outdated. 

When `stateProofWorker` starts for the second time, the same object is being set here.
There is nothing wrong here, but redundant work is being done. 

Moreover, the locks are held here, likely to avoid a race condition with `OnPrepareVoterCommit`.
Too many possibilities when allowing `OnPrepareVoterCommit` to execute as the worker is getting initialized. More on this, see my comment on `OnPrepareVoterCommit` in builder.go

I think it will be better to de-register the commit listener when stopping the worker, and register again when starting.